source("02_process_data.R")

#salmonbin <- "/usr/local/software/SalmonBeta-0.6.1_DebianSqueeze/bin/salmon"
#salmonbin <- "software/Salmon-0.7.2_linux_x86_64/bin/salmon"
salmonbin <- "/home/charlotte/software/Salmon-0.8.2_linux_x86_64/bin/salmon"
rapmapbin <- "/home/charlotte/software/RapMap/bin/rapmap"
fastqcbin <- "FastQC_v0.11.6.devel/fastqc"
cutadaptbin <- "/home/charlotte/miniconda3/bin/cutadapt"
#multiqcbin <- "/home/charlotte/miniconda2/bin/multiqc"
multiqcbin <- "/home/charlotte/miniconda3/bin/multiqc"

Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap <- 
  readRDS("reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap.rds")
Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap <- 
  readRDS("reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap.rds")
Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.txgenemap <- 
  readRDS("reference-files/danio-rerio/Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.txgenemap.rds")

Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges <- 
  readRDS("reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges.rds")
Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges <- 
  readRDS("reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges.rds")
Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.granges <- 
  readRDS("reference-files/danio-rerio/Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.granges.rds")

adapterseq <- "AGATCGGAAGAGC"
#adapterseq <- "file:adapters.fa"

## To run
## GSE84465
process_data(id = "GSE84465", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 29091775, shortname = "Darmanis2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84465",
             description = "glioblastoma tumors", 
             protocol = "Smart-Seq2", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("description", "neoplastic:ch1"),
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = 2, lps = "bottom", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## To run
## GSE81903
process_data(id = "GSE81903", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 27565351, shortname = "Shekhar2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81903",
             description = "P17 retinal cells from the Kcng4-cre;stop-YFP X Thy1-stop-YFP Line#1 mice", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.2", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## To run
## GSE80232
process_data(id = "GSE80232", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 27565351, shortname = "Shekhar2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80232",
             description = "P17 retinal cells from the Tg(Chx10-EGFP/cre,-ALPP)2Clc or Vsx2-GFP transgenic line", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("source_name_ch1", "characteristics_ch1.1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE80032
process_data(id = "GSE80032", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 27923766, shortname = "Wollny2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80032",
             description = "heterogeneity of acinar cells", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE94383
process_data(id = "GSE94383", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 28396000, shortname = "Lane2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE94383",
             description = "LPS stimulated and unstimulated 264.7 cells", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("characteristics_ch1.1", "characteristics_ch1.2"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE79102
process_data(id = "GSE79102", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 28346451, shortname = "Kiselev2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE79102",
             description = "different patients with myeloproloferative disease", 
             protocol = "Smart-Seq2", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1",
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = 2, lps = "bottom", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)


## To run
## GSE102299-smartseq2
process_data(id = "GSE102299-smartseq2", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 28902842, shortname = "Wallrapp2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE102299",
             description = "innate lymphoid cells from mouse lungs after various treatments", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.4", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE81076-GPL16791
process_data(id = "GSE81076-GPL16791", dtype = "umi", rtype = "paired", 
             organism = "Homo sapiens", genome = "GRCh38.84", 
             pmid = 27345837, shortname = "Grun2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81076", 
             description = "pancreatic cells from donors with and without type 2 diabetes",
             protocol = "CEL-Seq", protocoltype = "UMI", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq/transform.json", 
             cell_barcodes = "data-raw/GSE81076-GPL16791/GSE81076_cel-seq_barcodes.txt",
             sncol = "SampleName", groupid = "characteristics_ch1",
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE81076-GPL18573
process_data(id = "GSE81076-GPL18573", dtype = "umi", rtype = "paired", 
             organism = "Homo sapiens", genome = "GRCh38.84", 
             pmid = 27345837, shortname = "Grun2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81076", 
             description = "pancreatic cells from donors with and without type 2 diabetes",
             protocol = "CEL-Seq", protocoltype = "UMI", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq/transform.json", 
             cell_barcodes = "data-raw/GSE81076-GPL18573/GSE81076_cel-seq_barcodes.txt",
             sncol = "SampleName", groupid = "characteristics_ch1",
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE100911
process_data(id = "GSE100911", dtype = "fl", rtype = "paired",
             organism = "Danio rerio", genome = "GRCz10.87",
             pmid = "", shortname = "Tang2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE100911",
             description = "hematopoietic and renal cell heterogeneity", 
             protocol = "Smart-Seq2", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/danio-rerio/Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Danio_rerio.GRCz10.87.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)


## Not single cells
## GSE97142
process_data(id = "GSE97142", dtype = "umi", rtype = "paired", 
             organism = "Homo sapiens", genome = "GRCh38.84", 
             pmid = 28536247, shortname = "Dierickx2017", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE97142", 
             description = "", protocol = "", protocoltype = "", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq/transform.json", 
             cell_barcodes = "data-raw/GSE97142/GSE97142_cel-seq_barcodes.txt",
             sncol = "SampleName", groupid = "source_name_ch1",
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE71982
process_data(id = "GSE71982", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 26469390, shortname = "Burns2015", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71982",
             description = "utricular and cochlear sensory epithelia of newborn mice", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("source_name_ch1", "characteristics_ch1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE64016
process_data(id = "GSE64016", dtype = "fl", rtype = "single",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 26301841, shortname = "Leng2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE64016",
             description = "H1 and H1-Fucci cells", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1",
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = 2, lps = "bottom", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE66053-GPL18573
process_data(id = "GSE66053-GPL18573", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 25866248, shortname = "PadovanMerhar2015", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66053",
             description = "live and fixed single cells", 
             protocol = "Fluidigm C1 Auto prep", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.k15.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1",
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)


## SRP073808
process_data(id = "SRP073808", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 27996962, shortname = "Koh2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/sra?term=SRP073808",
             description = "in vitro cultured H7 embryonic stem cells (WiCell) and H7-derived downstream early mesoderm progenitors", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "Run", groupid = "LibraryName",
             geodata = FALSE, phenofile = "data-raw/SRP073808/SRP073808_pheno.txt",
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE52583-GPL16417
process_data(id = "GSE52583-GPL16417", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24739965, shortname = "Treutlein2014", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52583",
             description = "lung epithelial cells at different developmental stages", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin,  multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.2", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges, 
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges, 
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap, 
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE52583-GPL13112
process_data(id = "GSE52583-GPL13112", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24739965, shortname = "Treutlein2014", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52583",
             description = "lung epithelial cells at different developmental stages", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin,  multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.2", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges, 
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges, 
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap, 
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE62270-GPL19057
process_data(id = "GSE62270-GPL19057", dtype = "umi", rtype = "paired", 
             organism = "Mus musculus", genome = "GRCm38.84", 
             pmid = 26287467, shortname = "Grun2015", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62270", 
             description = "cells from mouse intestinal organoids", 
             protocol = "CEL-Seq", protocoltype = "UMI", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq/transform.json", 
             cell_barcodes = "data-raw/GSE62270-GPL19057/GSE62270_96BC.txt",
             sncol = "SampleName", groupid = "source_name_ch1",
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges, 
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges, 
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap, 
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE62270-GPL17021
process_data(id = "GSE62270-GPL17021", dtype = "umi", rtype = "paired", 
             organism = "Mus musculus", genome = "GRCm38.84", 
             pmid = 26287467, shortname = "Grun2015", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62270", 
             description = "cells from mouse intestinal organoids", 
             protocol = "CEL-Seq", protocoltype = "UMI", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq/transform.json", 
             cell_barcodes = "data-raw/GSE62270-GPL17021/GSE62270_96BC.txt",
             sncol = "SampleName", groupid = "source_name_ch1",
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges, 
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges, 
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap, 
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE)
  
## GSE78779-GPL17021
process_data(id = "GSE78779-GPL17021", dtype = "umi", rtype = "single", 
             organism = "Mus musculus", genome = "GRCm38.84", 
             pmid = 27121950, shortname = "Hashimshony2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE78779", 
             description = "dendritic and fibroblast cells analyzed with different protocols", 
             protocol = "CEL-Seq/CEL-Seq2/Smart-Seq", protocoltype = "UMI", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL,
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin, salmonindex = NULL, libtype = NULL, bias = FALSE, 
             rapmapbin = rapmapbin, 
             rapmapindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.5.0.ridx", 
             umis_transform = "software/umis/examples/CEL-Seq2/transform.json", 
             cell_barcodes = NULL, 
             sncol = "SampleName", groupid = c("source_name_ch1", "description"),
             geodata = TRUE, phenofile = NULL, 
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges, 
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges, 
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap, 
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "umis", "multiqc", "mae", "scater"),
             force = FALSE) 

## E-MTAB-3929
process_data(id = "EMTAB3929", dtype = "fl", rtype = "single",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 27062923, shortname = "Petropoulos2016", 
             datalink = "https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3929/",
             description = "preimplantation embryos from embryonic day 3 to 7", 
             protocol = "Smart-Seq2", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.8.2.sidx",
             libtype = "A", bias = TRUE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "LibraryName",
             groupid = c("Characteristics.developmental.stage.", "Characteristics.treatment."),
             geodata = FALSE, phenofile = "data-raw/EMTAB3929/EMTAB3929_pheno.txt",
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE66507
process_data(id = "GSE66507", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 26293300, shortname = "Blakeley2015", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66507",
             description = "preimplantation embryos", 
             protocol = "SMARTer", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin,  multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.7.2.sidx",
             libtype = "A", bias = FALSE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right", 
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE71585, HiSeq2000
process_data(id = "GSE71585-GPL13112", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 26727548, shortname = "Tasic2016", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71585",
             description = "cell type identification in primary visual cortex", 
             protocol = "SMARTer", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.7.2.sidx",
             libtype = "A", bias = FALSE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE71585, HiSeq2500
process_data(id = "GSE71585-GPL17021", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 26727548, shortname = "Tasic2016", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE71585",
             description = "cell type identification in primary visual cortex", 
             protocol = "SMARTer", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.7.2.sidx",
             libtype = "A", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1.1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## E-MTAB-2805
process_data(id = "EMTAB2805", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 25599176, shortname = "Buettner2015", 
             datalink = "https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2805/",
             description = "mESC in different cell cycle stages", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "LibraryName", groupid = "cell_cycle_stage", 
             geodata = FALSE, phenofile = "data-raw/EMTAB2805/EMTAB2805_pheno.txt",
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE77847
process_data(id = "GSE77847", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 27016502, shortname = "Meyer2016", 
             datalink = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77847",
             description = "Dnmt3a loss of function in Flt3-ITD and Dnmt3a-mutant AML", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "characteristics_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"), 
             force = FALSE)

## GSE44183human
process_data(id = "GSE44183-GPL11154", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 23892778, shortname = "Xue2013", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE44183",
             description = "embryonal development from oocyte to morula", 
             protocol = "Tang2010", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "Sample_Name_s", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),  
             force = FALSE)

## GSE44183mouse
process_data(id = "GSE44183-GPL13112", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 23892778, shortname = "Xue2013", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE44183",
             description = "embryonal development from oocyte to morula", 
             protocol = "Tang2010", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "Sample_Name_s", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE44183mouse, trimmed
process_data(id = "GSE44183-GPL13112-trimmed", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 23892778, shortname = "Xue2013", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE44183",
             description = "embryonal development from oocyte to morula", 
             protocol = "Tang2010", protocoltype = "full-length",
             dotrim = TRUE, cutadaptbin = cutadaptbin, adapterseq = "AGATCGGAAGAGC", 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "Sample_Name_s", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE45719
process_data(id = "GSE45719", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24408435, shortname = "Deng2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45719",
             description = "development from zygote to blastocyst + adult liver", 
             protocol = "SMART-Seq", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin,  multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "U", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE74596
process_data(id = "GSE74596", dtype = "fl", rtype = "single",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 27089380, shortname = "Engel2016", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74596",
             description = "purified populations of thymic NKT cell subsets", 
             protocol = "Smart-Seq2", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "U", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".",
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE63818, HiSeq2500
process_data(id = "GSE63818-GPL16791", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 26046443, shortname = "Guo2015", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63818",
             description = "primordial germ cells from embryos at different times of gestation", 
             protocol = "Tang", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("source_name_ch1", "characteristics_ch1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = NULL, lps = "right",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE60749, HiSeq2000 (25bp reads)
process_data(id = "GSE60749-GPL13112", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 25471879, shortname = "Kumar2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60749",
             description = "mESCs with various genetic perturbations, cultured in different media", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("source_name_ch1", "characteristics_ch1.1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE60749, HiSeq2500 (25bp reads)
process_data(id = "GSE60749-GPL17021", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 25471879, shortname = "Kumar2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60749",
             description = "mESCs with various genetic perturbations, cultured in different media", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("characteristics_ch1", "characteristics_ch1.1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE57872 (25bp reads)
process_data(id = "GSE57872", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 24925914, shortname = "Patel2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57872",
             description = "glioblastoma cells from tumors + gliomasphere cell lines", 
             protocol = "SMART-Seq", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.k15.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = c("characteristics_ch1", "characteristics_ch1.1"), 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE41265
process_data(id = "GSE41265", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 23685454, shortname = "Shalek2013", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41265",
             description = "stimulated immune cells", 
             protocol = "SMARTer", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 1, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE52529-GPL11154
process_data(id = "GSE52529-GPL11154", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 24658644, shortname = "Trapnell2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52529",
             description = "primary myoblasts over a time course of serum-induced differentiation", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE52529-GPL16791
process_data(id = "GSE52529-GPL16791", dtype = "fl", rtype = "paired",
             organism = "Homo sapiens", genome = "GRCh38.84",
             pmid = 24658644, shortname = "Trapnell2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52529",
             description = "primary myoblasts over a time course of serum-induced differentiation", 
             protocol = "SMARTer C1", protocoltype = "full-length", 
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/homo-sapiens/Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Homo_sapiens.GRCh38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE48968-GPL13112
process_data(id = "GSE48968-GPL13112", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24919153, shortname = "Shalek2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48968",
             description = "dendritic cells stimulated with pathogenic components", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE48968-GPL17021-125bp
process_data(id = "GSE48968-GPL17021-125bp", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24919153, shortname = "Shalek2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48968",
             description = "dendritic cells stimulated with pathogenic components", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.0.6.1.sidx",
             libtype = "IU", bias = FALSE,
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)

## GSE48968-GPL17021-25bp
process_data(id = "GSE48968-GPL17021-25bp", dtype = "fl", rtype = "paired",
             organism = "Mus musculus", genome = "GRCm38.84",
             pmid = 24919153, shortname = "Shalek2014", 
             datalink = "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48968",
             description = "dendritic cells stimulated with pathogenic components", 
             protocol = "SMARTer C1", protocoltype = "full-length",
             dotrim = FALSE, cutadaptbin = cutadaptbin, adapterseq = NULL, 
             fastqcbin = fastqcbin, multiqcbin = multiqcbin, 
             salmonbin = salmonbin,
             salmonindex = "reference-files/mus-musculus/Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.k15.0.6.1.sidx",
             libtype = "IU", bias = FALSE, 
             rapmapbin = rapmapbin, rapmapindex = NULL, umis_transform = NULL, cell_barcodes = NULL,
             sncol = "SampleName", groupid = "source_name_ch1", 
             geodata = TRUE, phenofile = NULL,
             gene_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$gene_granges,
             tx_granges = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.granges$tx_granges,
             txgenemap = Mus_musculus.GRCm38.84.cdna.ncrna.ercc92.txgenemap,
             tmpdir = "tmp", topdir = ".", 
             nrw = 3, lps = "bottom",
             aspects = c("fastqc", "salmon", "multiqc", "mae", "scater"),
             force = FALSE)
